File: ComplementApp.API/Models/CDP.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace ComplementApp.API.Models
{
    [Table("TCDP")]
    public class CDP
    {
        public int CdpId { get; set; }

        public int Instancia { get; set; }

        public long Cdp { get; set; }

        public long Crp { get; set; }

        public long Obligacion { get; set; }

        public long OrdenPago { get; set; }

        public DateTime Fecha { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle1 { get; set; }

        [Column(TypeName = "decimal(30,8)")]
        public decimal ValorInicial { get; set; }

        [Column(TypeName = "decimal(30,8)")]
        public decimal Operacion { get; set; }

        [Column(TypeName = "decimal(30,8)")]
        public decimal ValorTotal { get; set; }

        [Column(TypeName = "decimal(30,8)")]
        public decimal SaldoActual { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle2 { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle3 { get; set; }

        [Column(TypeName = "VARCHAR(4000)")]
        public string Detalle4 { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle5 { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle6 { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle7 { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle8 { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle9 { get; set; }

        [Column(TypeName = "VARCHAR(250)")]
        public string Detalle10 { get; set; }

        public int RubroPresupuestalId { get; set; }
        public RubroPresupuestal RubroPresupuestal { get; set; }
        public int TerceroId { get; set; }

        public Tercero Tercero { get; set; }

    }
}

File: ComplementApp-SPA/src/app/_models/cdp.ts
export interface Cdp {
  id: number;
  cdpId: number;
  cdp: number;
  crp: number;
  ordenPago: number;
  obligacion: number;
  numeroDocumento: number;

  fecha: Date;
  fechaFormato: string;
  detalle1: string; // Estado
  detalle2: string; // Codigo dependencia
  detalle4: string; // Objeto del bien
  objeto: string; // Objeto del bien completo
  detalle5: string; // Supervisor
  detalle6: string; // Numero Contrado
  detalle7: string; // ModalidadContrato
  detalle8: string; // Fuente Financiacion FTE
  detalle9: string; // Situacion Fondo SIT
  detalle10: string; // Recurso Presupuestal REC
  supervisorId: number;
  terceroId: number;

  valorInicial: number;
  operacion: number;
  valorTotal: number;
  saldoActual: number;

  numeroIdentificacionTercero: string;
  nombreTercero: string;
  identificacionRubro: string;
  nombreRubro: string;
  pci: string;
  cantidadMaxima: number;
  valorPagadoFechaActual: number;
  numeroPagoFechaActual: number;

  formatoSolicitudPagoId: number;
  planPagoId: number;
  contratoId?: number;
  esSeleccionada: boolean;
  valorFacturado: number;
}
